services/filesstore/s3store.go

```diff
@@ -229,6 +230,65 @@ func (b *S3FileBackend) WriteFile(fr io.Reader, path string) (int64, *model.AppE
 	return info.Size, nil
 }
 
+func (b *S3FileBackend) AppendFile(fr io.Reader, path string) (int64, *model.AppError) {
+	s3Clnt, err := b.s3New()
+	if err != nil {
+		return 0, model.NewAppError("AppendFile", "api.file.append_file.s3.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	fp := filepath.Join(b.pathPrefix, path)
+	if _, err = s3Clnt.StatObject(context.Background(), b.bucket, fp, s3.StatObjectOptions{}); err != nil {
+		return 0, model.NewAppError("AppendFile", "api.file.append_file.s3.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	var contentType string
+	if ext := filepath.Ext(fp); model.IsFileExtImage(ext) {
+		contentType = model.GetImageMimeType(ext)
+	} else {
+		contentType = "binary/octet-stream"
+	}
+
+	var sse encrypt.ServerSide
+	if b.encrypt {
+		sse = encrypt.NewSSE()
+	}
+	options := s3.PutObjectOptions{
+		ContentType:          contentType,
+		ServerSideEncryption: sse,
+	}
+
+	partName := fp + ".part"
+	info, err := s3Clnt.PutObject(context.Background(), b.bucket, partName, fr, -1, options)
+	defer s3Clnt.RemoveObject(context.Background(), b.bucket, partName, s3.RemoveObjectOptions{})
```

Comment:
Should these two share the same context?

Reply:
The context parameter was an addition introduced by v7 of minio (https://github.com/mattermost/mattermost-server/pull/15177). 
We are not currently making use of it so I would say it doesn't make a difference at this point. 
If/when we decide to leverage that then a common context should be provided to both methods.